pkg/webhooks/controlplanemachineset/webhooks_test.go

```diff
@@ -84,118 +89,155 @@ var _ = Describe("Webhooks", func() {
 		var machineTemplate resourcebuilder.OpenShiftMachineV1Beta1TemplateBuilder
 
-		BeforeEach(func() {
-			providerSpec := resourcebuilder.AWSProviderSpec().WithAvailabilityZone("us-east-1")
-			machineTemplate = resourcebuilder.OpenShiftMachineV1Beta1Template().WithProviderSpecBuilder(providerSpec)
-			// Default CPMS builder should be valid, individual tests will override to make it invalid
-			builder = resourcebuilder.ControlPlaneMachineSet().WithNamespace(namespaceName).WithMachineTemplateBuilder(machineTemplate)
-		})
-
-		It("with a valid spec", func() {
-			cpms := builder.Build()
-			Expect(k8sClient.Create(ctx, cpms)).To(Succeed())
-		})
+		Context("when validating without without failure domains", func() {
+			BeforeEach(func() {
+				providerSpec := resourcebuilder.AWSProviderSpec().WithAvailabilityZone("us-east-1")
+				machineTemplate = resourcebuilder.OpenShiftMachineV1Beta1Template().WithProviderSpecBuilder(providerSpec)
+				// Default CPMS builder should be valid, individual tests will override to make it invalid
+				builder = resourcebuilder.ControlPlaneMachineSet().WithNamespace(namespaceName).WithMachineTemplateBuilder(machineTemplate)
 
-		PIt("with a disallowed name", func() {
-			cpms := builder.WithName("disallowed").Build()
-			Expect(k8sClient.Create(ctx, cpms)).To(MatchError(ContainSubstring("TODO")))
-		})
+				machineBuilder := resourcebuilder.Machine().WithNamespace(namespaceName)
+				controlPlaneMachineBuilder := machineBuilder.WithGenerateName("control-plane-machine-").AsMaster().WithProviderSpecBuilder(providerSpec)
+				By("Creating a selection of Machines")
+				for i := 0; i < 3; i++ {
+					controlPlaneMachine := controlPlaneMachineBuilder.Build()
+					Expect(k8sClient.Create(ctx, controlPlaneMachine)).To(Succeed())
+				}
+			})
 
-		It("with 4 replicas", func() {
-			cpms := builder.WithReplicas(4).Build()
-			Expect(k8sClient.Create(ctx, cpms)).To(MatchError(ContainSubstring("Unsupported value: 4: supported values: \"3\", \"5\"")))
-		})
+			It("with a valid spec", func() {
+				cpms := builder.Build()
+				Expect(k8sClient.Create(ctx, cpms)).To(Succeed())
+			})
 
-		PIt("with mismatched selector and machine labels", func() {
-			cpms := builder.WithSelector(metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					openshiftMachineRoleLabel:            masterMachineRole,
-					openshiftMachineTypeLabel:            masterMachineRole,
-					machinev1beta1.MachineClusterIDLabel: "cluster-id",
-				},
-			}).WithMachineTemplateBuilder(
-				machineTemplate.WithLabels(map[string]string{
-					openshiftMachineRoleLabel:            masterMachineRole,
-					openshiftMachineTypeLabel:            masterMachineRole,
-					machinev1beta1.MachineClusterIDLabel: "different-id",
-				}),
-			).Build()
+			It("with a disallowed name", func() {
+				cpms := builder.WithName("disallowed").Build()
+				Expect(k8sClient.Create(ctx, cpms)).To(MatchError(ContainSubstring(errControlPlaneInvalidName.Error())))
```

Comment:
This _should_ work, does it error?
```suggestion
				Expect(k8sClient.Create(ctx, cpms)).To(MatchError(errControlPlaneInvalidName))
```